### PR TITLE
UCP: Fix behaviour of tag-match protocol in case am.max_inline is 0.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/574/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3032/ for details.

**Comment by shamisp** [comment]
is this the fix for LANL ?

File: src/ucp/core/ucp_ep.h [hunk 1]
```diff
@@ -84,8 +84,8 @@ typedef struct ucp_ep_config {
 
     /* Limits for active-message based protocols */
     struct {
-        size_t                 max_eager_short;  /* Maximal payload of eager short */
-        size_t                 max_short;        /* Maximal payload of am short */
+        ssize_t                max_eager_short;  /* Maximal payload of eager short */
+        ssize_t                max_short;        /* Maximal payload of am short */
```
Comment (shamisp) [inline src/ucp/core/ucp_ep.h]: do we need to change the rest to ssize as well, so in future we have an option to bypass it ?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/576/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2516/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp we would probably have to fix others as well. 
It's could also help to fix LANL issue, but probably not enough.

**Comment by brminich** [comment]
if RC transports are relevant, we would need to disable FC if AM short is not supported (or update FC to use non-inline protocols for auxiliary messages)

**Comment by shamisp** [comment]
LANL folks wanted to disable short , seems like this should do it, right ?

**Comment by yosefe** [comment]
For tag-match on UCP, yes. maybe more fixes are required for RMA, and on UCT layer.

**Pull request merged**